### PR TITLE
RATIS-1570. Add getLeaderId method to DivisionInfo

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
see [jira1570](https://issues.apache.org/jira/browse/RATIS-1570)

### CODE REVIEW

**Comment by OneSizeFitsQuorum** [comment]
@szetszwo PTAL

File: ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java [hunk 1]
```diff
@@ -51,6 +52,10 @@ default boolean isListener() {
   /** Is this server division currently the leader and ready? */
   boolean isLeaderReady();
 
+  /** @return the current leader of this server division, null means either there is
+   * no leader for this term yet or this server division does not know who it is yet */
+  RaftPeer getCurrentLeader();
```
Comment (szetszwo) [inline ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java:57]: Let's return RaftPeerId so that it is more efficient?  When RaftPeer is needed, we may call RaftServer.getRaftConf() to get it.
```
  /**
   * @return the id of the current leader if the leader is known to this server division;
   *         otherwise, return null.
   */
  RaftPeerId getLeaderId();
```

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java [hunk 2]
```diff
@@ -113,9 +113,14 @@ public boolean isLeaderReady() {
...
 
+    @Override
+    public RaftPeer getCurrentLeader() {
+      return getRaftConf().getPeer(getState().getLeaderId());
+    }
+
     @Override
     public LifeCycle.State getLifeCycleState() {
-      return lifeCycle.getCurrentState();
+      return getLifeCycle().getCurrentState();
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java:123]: Please revert this and the following unrelated changes.

**Review by szetszwo — CHANGES_REQUESTED** [review]
@OneSizeFitsQuorum , thanks for working on this!  Please see the comments inlined.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java (hunk 2 above)
Comment (OneSizeFitsQuorum) [reply ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java]: Fixed

File: ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java]: Fixed

File: ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java [hunk 3]
```diff
@@ -20,6 +20,8 @@
 
 import org.apache.ratis.proto.RaftProtos.RaftPeerRole;
 import org.apache.ratis.proto.RaftProtos.RoleInfoProto;
+import org.apache.ratis.protocol.RaftPeer;
```
Comment (szetszwo) [inline ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java:23]: Please remove this.  There is a checkstyle warning: https://github.com/apache/ratis/runs/6335329880?check_suite_focus=true#step:4:262 .

File: ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java [hunk 4]
```diff
@@ -51,6 +53,12 @@ default boolean isListener() {
   /** Is this server division currently the leader and ready? */
   boolean isLeaderReady();
 
+  /**
+   * @return return the id of the current leader if the leader is known to this server division;
```
Comment (szetszwo) [inline ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java:57]: The second "return" should be removed; see https://www.oracle.com/hk/technical-resources/articles/java/javadoc-tool.html

**Review by szetszwo — COMMENTED** [review]
@OneSizeFitsQuorum , thanks for the update.  Two minor comments inlined.

File: ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java (hunk 4 above)
Comment (OneSizeFitsQuorum) [reply ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java]: Fixed

File: ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java (hunk 3 above)
Comment (OneSizeFitsQuorum) [reply ratis-server-api/src/main/java/org/apache/ratis/server/DivisionInfo.java]: Sorry because the Google CheckStyle I imported locally seems to be different from your style tool, so I did not format the code, so this line dependency is not removed

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.
